runtime/drivers/clickhouse/model_executor_s3_self.go

```diff
@@ -80,6 +80,22 @@ func (e *s3ToSelfExecutor) Execute(ctx context.Context, opts *drivers.ModelExecu
 	clone.InputProperties = propsMap
 	newOpts := &clone
 
+	// set output props with defaults
+	out := &ModelOutputProperties{}
+	if err := mapstructure.WeakDecode(newOpts.OutputProperties, out); err != nil {
+		return nil, fmt.Errorf("failed to parse output properties: %w", err)
+	}
+	if out.Materialize == nil {
+		materialize := true
+		out.Materialize = &materialize
+	} else if !*out.Materialize {
+		return nil, fmt.Errorf("s3 to clickhouse executor only supports materialized models. Set `materialize` to true")
+	}
+	if err := mapstructure.WeakDecode(out, &newOpts.OutputProperties); err != nil {
+		return nil, fmt.Errorf("failed to parse output properties: %w", err)
+	}
```

Comment:
The DuckDB executors apply defaults when consuming the output properties (when the `Validate` func is called). We could do the same here? 

See: https://github.com/rilldata/rill/blob/d66a28ff94b55273d102cc2112faf8d05465d254/runtime/drivers/duckdb/model_executor_self.go#L42-L44

And: https://github.com/rilldata/rill/blob/d66a28ff94b55273d102cc2112faf8d05465d254/runtime/drivers/duckdb/model_manager.go#L31-L44

Reply:
So this validation is only specific to `s3_clickhouse` model and not to all.
Added a specific property that embeds and doing validation on that.

Reply:
What do you mean it's only specific to that executor? Seems to me like the default value for `materialize` should always be `true` when the output is a known physical destination (like S3 or incremental table), like in DuckDB?

Reply:
I don't understand why we can't apply the same defaults as DuckDB in all the cases? I'm not thinking about always setting it to `true`, I'm thinking about these defaults linked above (i.e. defaulting to true for incremental models, split models, and multi-connector models): https://github.com/rilldata/rill/blob/d66a28ff94b55273d102cc2112faf8d05465d254/runtime/drivers/duckdb/model_manager.go#L31-L44

Reply:
Ahh okay. I understand now. Yeah I think this makes sense.